oauthproxy.go

```diff
@@ -634,7 +751,7 @@ func getRemoteAddr(req *http.Request) (s string) {
 }
 
 func (p *OAuthProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	switch path := req.URL.Path; {
+	switch path := req.URL.EscapedPath(); {
```

Comment:
This is necessary to support mapping escaped paths. To your point, though, I think it would likely fail to match because the configuration doesn't also escape the ping path (or any configured "internal" path) during options validation. I'm not sure you *want* to do that though, as people should be providing the path pre-escaped (i.e. a URL path, not a file path), and you don't want to double-escape it. It keeps things conceptually all in the realm of URLs and avoids forcing people to wonder which option takes what (file path or URL path).

Reply:
I think your approach is best. Given that it's unlikely that someone were to use characters that need escaping I think this is fine. Thanks for the explanation.